src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -483,18 +484,34 @@ public AzureApplicationDeploymentResource getApplicationDeploymentResource(UUID
 
   public void grantPoliciesBqJobUser(String dataProject, Collection<String> policyEmails)
       throws InterruptedException {
-    final List<String> emails =
-        policyEmails.stream().map((e) -> "group:" + e).collect(Collectors.toList());
-    resourceManagerService.updateIamPermissions(
-        Collections.singletonMap(BQ_JOB_USER_ROLE, emails), dataProject, ENABLE_PERMISSIONS);
+    modifyRoles(dataProject, policyEmails, BQ_JOB_USER_ROLE, ENABLE_PERMISSIONS);
   }
 
   public void revokePoliciesBqJobUser(String dataProject, Collection<String> policyEmails)
       throws InterruptedException {
+    modifyRoles(dataProject, policyEmails, BQ_JOB_USER_ROLE, REVOKE_PERMISSIONS);
+  }
+
+  public void grantStorageObjectViewer(String dataProject, Collection<String> policyEmails)
+      throws InterruptedException {
+    modifyRoles(dataProject, policyEmails, STORAGE_OBJECT_VIEWER_ROLE, ENABLE_PERMISSIONS);
+  }
+
+  public void revokeStorageObjectViewer(String dataProject, Collection<String> policyEmails)
+      throws InterruptedException {
+    modifyRoles(dataProject, policyEmails, STORAGE_OBJECT_VIEWER_ROLE, REVOKE_PERMISSIONS);
+  }
```

Comment:
Unless I'm misunderstanding the code, I think this needs to change to modify the permissions directly on the bucket, not for the general IAM of the google data project.

Reply:
Yeah, I agree. It's a little different than the BQ job permissions which are scoped to project.  I think this should be on the data bucket (we don't want to give access to scratch)